tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/json/FirebaseSubscriptionSerializer.scala

```diff
@@ -35,15 +39,53 @@ object FirebaseSubscriptionSerializer {
         "expires" -> response.expires,
         "types" -> response.types)
 
+  private implicit val mapCreationRequestBySubscriptionCreationId: Reads[Map[FirebaseSubscriptionCreationId, JsObject]] =
+    Reads.mapReads[FirebaseSubscriptionCreationId, JsObject] { string =>
+      refineV[IdConstraint](string)
+        .fold(e => JsError(s"firebase subscription id needs to match id constraints: $e"),
+          id => JsSuccess(FirebaseSubscriptionCreationId(id)))
+    }
+
+  private implicit val patchObject: Reads[FirebaseSubscriptionPatchObject] = Json.valueReads[FirebaseSubscriptionPatchObject]
+
+  private implicit val mapUpdateRequestBySubscriptionCreationId: Reads[Map[UnparsedFirebaseSubscriptionId, FirebaseSubscriptionPatchObject]] =
```

Comment:
I don't think this work is about set update yet :)

Reply:
Or we can let Tung crashes the sprint :p